File: src/main/java/com/godwin/jsonparser/ui/action/AddTabAction.java
package com.godwin.jsonparser.ui.action;

import com.godwin.jsonparser.ui.IParserWidget;
import com.intellij.icons.AllIcons;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import org.jetbrains.annotations.NotNull;

/**
 * Created by fingerart on 17/2/20.
 */
public class AddTabAction extends AnAction {

    private final IParserWidget mParserWidget;

    public AddTabAction(IParserWidget parserWidget) {
        super("Add Tab", "Create New ApiDebugger Tab", AllIcons.General.Add);
        mParserWidget = parserWidget;
    }

    @Override
    public void actionPerformed(AnActionEvent e) {
        mParserWidget.createParserSession();
        System.out.println("AddTabAction.actionPerformed");
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        super.update(e);
        e.getPresentation().setVisible(mParserWidget.getTabCount() <= 10);
    }
}

File: src/main/kotlin/com/godwin/jsonparser/action/AddTabAction.kt
package com.godwin.jsonparser.action

import com.godwin.jsonparser.ui.IParserWidget
import com.godwin.jsonparser.util.analytics.Analytics
import com.godwin.jsonparser.util.analytics.AnalyticsConstant
import com.intellij.icons.AllIcons
import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent

class AddTabAction(private val parserWidget: IParserWidget) : AnAction(
    "Add Tab",
    "Create new parser tab",
    AllIcons.General.Add
) {
    override fun actionPerformed(e: AnActionEvent) {
        Analytics.track(AnalyticsConstant.ACTION_ADD_TAB)
        parserWidget.createParserSession()
    }

    override fun update(e: AnActionEvent) {
        super.update(e)
        e.presentation.isVisible = parserWidget.getTabCount() <= 10
    }

    override fun getActionUpdateThread() = ActionUpdateThread.BGT
}
